src/listpack.c

```diff
@@ -996,13 +996,39 @@ unsigned char *lpDelete(unsigned char *lp, unsigned char *p, unsigned char **new
     return lpInsert(lp,NULL,NULL,0,p,LP_REPLACE,newp);
 }
 
+/* Delete a range of entries from the first address to tail address. 
+ * 'num' means the num of entries in this range. The function returns 
+ * the new listpack as return value, and also sets 'first' pointing to the next entry. */
+unsigned char *lpDeleteRangeWithEntryPtr(unsigned char *lp, unsigned char **first, unsigned char *tail, unsigned long num) {
+    if (num == 0) return lp;  /* Nothing to delete, return ASAP. */
+    size_t bytes = lpBytes(lp);
+    unsigned char *eofptr = lp + bytes - 1;
+    assert(*first != NULL); 
+    if (tail == NULL) tail = eofptr;
+    /* Store the offset of the element 'first', so that we can obtain its
+     * address again after a reallocation. */
+    unsigned long poff = *first-lp;
+
+    /* Move tail to the front of the listpack */
+    memmove(*first, tail, eofptr - tail + 1);
```

Comment:
i'm worried about about this, if the `bytes` is corrupted, does this mean it's possible to use eofptr to read the memory data after this listpack?

Reply:
I guess that one reason of the performance improvement is the reduction calls of `lpAssertValidEntry`.

Reply:
In `lpDeleteRangeWithEntry`, if the `bytes` is corrupted, we also have the same problem. If we want to avoid this problem, I think maybe we have to travel the whole listpack to verify the validity of the `bytes`. 

We get `first` and `tail` by `lpNext` or `lpPrev`, in which we have called `lpAssertValidEntry`. I think maybe we don't need to repeat it in `lpDeleteRangeWithEntry`.